### PR TITLE
Fix PageSearch scrollbar

### LINKED ISSUE #168: Cannot drag scroll tab search scrollbar
afbeelding

Cannot drag scroll tab search scrollbar.

### PR DESCRIPTION
**Pull request by veger**
`gui.SetTextInputFocus` was setting `gui.mouseDownRect`, which is heavily used by scrollbars to detect is the mouse is still inside the scrollbar rect when dragging.

Since the `PageSearch` focus gets updated every (build) event, the `mouseDownRect` was effectively always set to the search input, and the scrollbar could not be dragged as it detected that the mouse was not in the correct `Rect` anymore.

I fixes the issue by introducing a `gui.focusRect`, so these `Rect`s are tracked independently. Both typing text, as well as dragging the `PageSearch` scrollbar seem to be working now.

I (quickly) checked some other input fields, but did not see any regressions.

(The other commit is a small cleanup that I encountered while debugging this issue)

Fixes #168

### CODE REVIEW

File: Yafc.UI/ImGui/ImGuiBuilding.cs [hunk 1]
```diff
@@ -171,6 +171,7 @@ public partial class ImGui {
         public bool mousePresent { get; private set; }
         public Rect mouseDownRect { get; private set; }
         public Rect mouseOverRect { get; private set; } = Rect.VeryBig;
+        public Rect focusRect { get; private set; }
```
Comment (have-fun-was-taken) [inline Yafc.UI/ImGui/ImGuiBuilding.cs:174]: Would be good to have a comment that explains the purpose of this variable.

Comment (veger) [reply Yafc.UI/ImGui/ImGuiBuilding.cs:175]: totally forgot, sorry

**Review by have-fun-was-taken — APPROVED** [review]

File: Yafc.UI/ImGui/ImGuiBuilding.cs [hunk 2]
```diff
@@ -349,14 +351,14 @@ public partial class ImGui {
...
         public void ClearFocus() {
-            if (mouseDownRect != default) {
-                mouseDownRect = default;
+            if (focusRect != default) {
+                focusRect = default;
                 Rebuild();
             }
         }
 
         public void SetFocus(Rect rect) {
```
Comment (DaleStan) [inline Yafc.UI/ImGui/ImGuiBuilding.cs:360]: The call to `SetFocus` doesn't immediately seem to need to be replaced by a call to `Rebuild`, but I know there are Windows/Linux differences there. I'd delete `SetFocus` either way.

**Review by DaleStan — CHANGES_REQUESTED** [review]

File: Yafc.UI/ImGui/ImGuiBuilding.cs [hunk 3]
```diff
@@ -349,14 +351,14 @@ public partial class ImGui {
         public bool IsLastMouseDown(Rect rect) => rect == mouseDownRect;
 
         public void ClearFocus() {
-            if (mouseDownRect != default) {
-                mouseDownRect = default;
+            if (focusRect != default) {
```
[the remark this answers is not in the document]
Comment (veger) [reply Yafc.UI/ImGui/ImGuiBuilding.cs]: Thanks, I did not realize these function were not/barely used. I fully removed `SetFocus()` and `ClearFocus()`.

File: Yafc.UI/ImGui/ImGuiBuilding.cs (hunk 2 above)
Comment (veger) [reply Yafc.UI/ImGui/ImGuiBuilding.cs]: I moved `Rebuild()` to `SetInputTextFocus()` just to be sure everything keeps working (these UI things are so/too easy to break, and we do not have any tests for them/their functionality...)

**Review by DaleStan — APPROVED** [review]

**Pull request merged**